File: backend/src/shared/enums.ts

export enum UserRole {
    DOCTOR = 'DOCTOR',
    PATIENT = 'PATIENT',
}

export enum BloodType {
    A_POS = 'A_POS',
    A_NEG = 'A_NEG',
    B_POS = 'B_POS',
    B_NEG = 'B_NEG',
    AB_POS = 'AB_POS',
    AB_NEG = 'AB_NEG',
    O_POS = 'O_POS',
    O_NEG = 'O_NEG',
}

export enum ConsultationStatus {
    SCHEDULED = 'SCHEDULED',
    COMPLETED = 'COMPLETED',
}

export enum VaccineStatus {
    PENDING = 'PENDING',
    DONE = 'DONE',
    LATE = 'LATE',
}

export enum ExamType {
    PDF = 'PDF',
    IMAGE = 'IMAGE',
}

export enum ExamStatus {
    UPLOADED = 'UPLOADED',
    REVIEWED = 'REVIEWED',
    REQUESTED = 'REQUESTED',
    REALIZED = 'REALIZED',
}

export enum AlertType {
    WARNING = 'WARNING',
    INFO = 'INFO',
    SUCCESS = 'SUCCESS',
}

export enum AlertTargetRole {
    DOCTOR = 'DOCTOR',
    PATIENT = 'PATIENT',
    BOTH = 'BOTH',
}

File: backend/Models/Enums.cs
using System.Text.Json.Serialization;

namespace backend.Models;

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum UserRole
{
    DOCTOR,
    PATIENT
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum BloodType
{
    A_POS,
    A_NEG,
    B_POS,
    B_NEG,
    AB_POS,
    AB_NEG,
    O_POS,
    O_NEG
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum ConsultationStatus
{
    SCHEDULED,
    COMPLETED
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum VaccineStatus
{
    PENDING,
    DONE,
    LATE
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum ExamType
{
    PDF,
    IMAGE
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum ExamStatus
{
    UPLOADED,
    REVIEWED,
    REQUESTED,
    REALIZED
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum AlertType
{
    WARNING,
    INFO,
    SUCCESS
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum AlertTargetRole
{
    DOCTOR,
    PATIENT,
    BOTH
}
